### ISSUE TITLE
Convert services to I/O via Streams

### ISSUE THREAD

**Issue by MarkAMartin** [eclipse/eavp#177]

**Comment by SmithRWORNL** [comment]
@jayjaybillings Currently, EAVP accepts input via post with jax rs @Consumes annotations controlling the content input type. The discussion before processing this issue is whether or not this meets the requirements for Stream input/output and if not what the requirements are.

**Comment by MarkAMartin** [comment]
Currently we want to use the OCTET Mime Type via JAX-RS StreamingOutput and StreamingInput.

**Comment by jayjaybillings** [comment]
Yes, we want to use application/octet-stream as the MIME type. We could do this with any particular RESTful service, but JAX-RS is always my go-to since it is so easy.

The other thing we discussed is that we should have build a small library that constructs the final product from the octet stream. This small library would then be a dependency for each microservice that needs it. This makes the data transfer very efficient and allows for flexibility in the marshalling and unmarshalling of data.

**Comment by jayjaybillings** [comment]
I have a strong preference for using JAX-RS since it is a standard and is very easy.

**Comment by SmithRWORNL** [comment]
This is done in the Robert/MicroServices branch.
